src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, Inject } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import {
+  WalletSelectionValue,
+} from 'src/app/common/components/mat-controls/transaction-type-mat-select/transaction-type-mat-select.component';
+import { WalletTransactionType } from 'src/app/domains/wallets/domains.wallets.types';
+import { ITransactionModalData, IWalletTransactionModalFormType } from './pages-wallet-transaction.editor.types';
+
+@Component({
+  selector: 'app-transaction-editor',
+  templateUrl: './pages-wallet-transaction-editor.component.html',
+  styleUrls: [ './pages-wallet-transaction-editor.component.scss' ],
+})
+export class PagesWalletTransactionEditorComponent {
+  public form!: FormGroup<IWalletTransactionModalFormType>;
+
+  public constructor(
+    private readonly dialogRef: MatDialogRef<PagesWalletTransactionEditorComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: Partial<ITransactionModalData>,
+  ) {
+    this.form = new FormGroup<IWalletTransactionModalFormType>({
+      amount: new FormControl(this.data.amount ?? 100, {
+        validators: [
+          Validators.required,
+          Validators.min(1),
+        ],
+        nonNullable: true,
+      }),
+      description: new FormControl(this.data.description ?? ''),
```

Comment:
Not sure if we should send empty string if description is optional, not send description at all or send `null` value. Maybe you can ask on `os-backend` channel about this? :)